### PR TITLE
[RFR] Game page rendering

### PR DESCRIPTION
**Pull request by JulienMattiussi**
https://trello.com/c/xV6wZFTP

Follw #1 

- [x] Get datas from API
- [x] Render grid
- [x] Render remaining pieces
- [x] Tests rendering
- [x] Refresh game every x time

### CODE REVIEW

File: pages/_document.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import Document, { Head, Main, NextScript } from 'next/document';
+import React from 'react';
+
+export default class MyDocument extends Document {
+    static async getInitialProps(ctx) {
+        const initialProps = await Document.getInitialProps(ctx);
```
Comment (ThieryMichel) [inline pages/_document.js]: return directly, no need to spread since you do not add props

File: pages/Game.js [hunk 2]
```diff
@@ -1,18 +1,103 @@
...
+        loaded: false,
+    };
+
+    static propTypes = {
+        game: PropTypes.object.isRequired,
+        loaded: PropTypes.bool,
+    };
+
+    static async getInitialProps() {
+        let url = `http://${config.apiUrl}/479`;
```
Comment (ThieryMichel) [inline pages/Game.js]: create a dataProvider with a getGame method, and use it to fetch all the needed resource. Make it isomorphic with isomorphic fetch, so you can use it here and server side in the socket logic.

File: server.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+const nextHandler = nextApp.getRequestHandler();
+
+const request = require('request');
+
+io.on('connection', socket => {
+    const idgame = { true: 479, false: 480 };
+    let i = true;
+    this.interval = setInterval(async () => {
+        let url = `http://${config.apiUrl}/${idgame[i]}`;
+        request(url, function(error, response, body) {
```
Comment (ThieryMichel) [inline server.js]: Use the dataProvider.getGame

File: src/game/ActionText.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+const getActionText = props => {
+    if (props.closed) {
+        return 'The game is over !!';
+    }
+    if (!props.locked) {
+        if (props.selectedPiece > 0) {
+            return 'Place your piece on the board';
+        }
+        return 'Choose a piece for your opponent';
+    } else if (props.watch_only) {
```
Comment (ThieryMichel) [inline src/game/ActionText.js]: just if the else is unneeded.

File: src/ui/Button.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+const Button = styled('button')`
+    display: inline-block;
+    box-sizing: content-box;
+    cursor: pointer;
+    padding: 10px 20px;
+    border: 1px solid #018dc4;
+    border-radius: 3px;
+    font: normal medium/normal Arial, Helvetica, sans-serif;
+    color: rgba(255, 255, 255, 0.9);
+    -o-text-overflow: clip;
```
Comment (ThieryMichel) [inline src/ui/Button.js]: remove opera specific css

**Review by ThieryMichel — CHANGES_REQUESTED** [review]

File: config/config.dist.js [hunk 6]
```diff
@@ -0,0 +1,5 @@
+export default {
+    //apiUrl: 'http://192.168.86.248/gameapi',
```
Comment (Luwangel) [inline config/config.dist.js]: Remove this line from the dist

File: pages/Game.js (hunk 2 above)
Comment (Luwangel) [inline pages/Game.js]: The url could be a constant defined at the root of the file (to avoid useless computation).

File: pages/Game.js [hunk 7]
```diff
@@ -1,18 +1,103 @@
...
+import PropTypes from 'prop-types';
+import fetch from 'isomorphic-unfetch';
+import io from 'socket.io-client';
+import styled from 'react-emotion';
+
+const BoardContainer = styled('div')`
+    height: 500px;
+`;
 
 class Game extends React.Component {
```
Comment (Luwangel) [inline pages/Game.js]: Use directly `Component`: `import React, { Component } from 'react';`

File: pages/Game.js [hunk 8]
```diff
@@ -1,18 +1,103 @@
...
+
+    static async getInitialProps() {
+        let url = `http://${config.apiUrl}/479`;
+
+        const res = await fetch(url);
+        const data = await res.json();
+
+        return {
+            game: data,
+            loaded: data.grid ? true : false,
```
Comment (Luwangel) [inline pages/Game.js]: You can cast to a boolean value: `!!data.grid`

**Review by Luwangel — CHANGES_REQUESTED** [review]

File: src/game/GridBox.js [hunk 9]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Box from './Box';
+
+const GridBox = props => (
+    <Box
+        enabled={props.enabled}
+        boxValue={props.boxValue}
+        label={'gridbox_x' + String(props.x) + '_y' + String(props.y)}
```
Comment (Kmaschta) [inline src/game/GridBox.js]: Take advantage of template strings: `gridbox_x${props.x}_y${props.y}`

File: src/game/RemainingBox.js [hunk 10]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Box from './Box';
+
+const RemainingBox = props => (
+    <Box
+        boxSize="40"
+        enabled={props.enabled}
+        boxValue={props.boxValue}
+        label={'remainingbox'}
```
Comment (Kmaschta) [inline src/game/RemainingBox.js]: When the prop is only a string you can avoid the brakets: `label="remainingbox"`

**Review by ThieryMichel — APPROVED** [review]

**Pull request merged**